openmetadata-service/src/main/java/org/openmetadata/service/resources/apps/AppResource.java

```diff
@@ -723,6 +724,13 @@ public Response delete(
       throw new IllegalArgumentException(
           CatalogExceptionMessage.systemEntityDeleteNotAllowed(app.getName(), "SystemApp"));
     }
+
+    if (app.getScheduleType().equals(ScheduleType.Scheduled)
```

Comment:
why are we doing this, in AppResource, such kind of calls should be delegated to a more generic function served by Native Application, we cannot have app specific behaviours handled like this in AppResource

Reply:
Noted.

Reply:
Provided the cleanup function in NativeApplication. Each app can have its own cleanup implementation. Overrode the cleanup fn in SlackApplication. Now, from AppResource I am calling the cleanup function on app uninstall.